eurec4a_env/decomposition/time.py

```diff
@@ -0,0 +1,47 @@
+import xarray as xr
+
+from ..source_data import get_halo_flight_legs
+
+
+def annotate_with_halo_flight_legs(ds, time="time", legtype="circle"):
```

Comment:
This is absolutely great and works for Levels-2 and 3. Recommendation for Level-4, since time is no longer a coordinate, we can assign the "segment kind" to the circles, by associating with the unique circle IDs, that are consistent between JOANNE Level-4 and the YAML files.

Reply:
I think I get this, but I'm not 100% sure 😄 Maybe this is something you could add? Would we just need a few lines to download the level 4 data and then modify the `annotate_with_halo_flight_legs` function? The way it's written now it *should* work with any dataset which has a time coordinate (so could for example be used on splitting up model data too)

Reply:
Yep. you are right. I think we can do this when Level-4 is ready. Mine was only a recommendation. I can modify the function later.  For me, this PR is good to go. 👍